pmlc/dialect/pxa/transforms/tile_accumulate.cc

```diff
@@ -42,7 +42,8 @@ AffineParallelOp tileAccumulations(AffineParallelOp op, bool skipTrivial) {
   bool anyNonAccum = false;
   for (size_t i = 0; i < ranges.size(); i++) {
     auto arg = op.getIVs()[i];
-    if (si.strides.count(arg)) {
+    bool stridesEmpty = !si.strides.size() && !anyNonAccum;
+    if (si.strides.count(arg) || stridesEmpty) {
```

Comment:
si.strides is a map, it represent all index variate which we may use in reduction op. but in some cases, such as max case. the si.strides is empty. reduction op don't use any index directly. so it didn't perform tile at all. this may cause race condition.

Reply:
Thanks, could please add the 2d max example to tile_accumulation.mlir? Not urgent, we can check and submit the fix to plaidml tomorrow.

Reply:
I think jeremy's change is good. it can fix this issue. I think maybe we should just close this pr.

but according to jeremy's method, it means that we always tile the reduction op. and we don't need to check  whether it really needs tile.
so after tile_accumulation pass now, there is always multi affine.parallel op.
though sometimes it would be like those:
affine.parallel (%arg2, %arg3) = (0, 0) to (3, 4) step (3, 4) 

I'm not sure whether this would have some bad effect on performance.